src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,46 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<T> getAll(Class<? extends T> type) {
+        List<? extends Machine> machines = new ArrayList<>();
+        if (Bulldozer.class == type) {
+            machines = new BulldozerProducer().get();
+            return (List<T>) machines;
+        } else if (type == Excavator.class) {
+            machines = new ExcavatorProducer().get();
+            return (List<T>) machines;
+        } else if (type == Truck.class) {
+            machines = new TruckProducer().get();
+            return (List<T>) machines;
+        } else {
+            return Collections.emptyList();
+        }
```

Comment:
That's a bit different from what I've proposed.
1. Inside if/elseif body, only overwrite the `machines`.
2. Cast and return `machines` after if/elseif statement - you've already initialized an empty array list, so `Collections.emptyList()` call is not necessary.